### DISCUSSION TITLE
Usage of `azureWorkItemId` only for pull request creation

### DISCUSSION THREAD

**Discussion by m-isenmann** [renovatebot/renovate#32119] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Azure

### Please tell us more about your question or problem

We are using the `azureWorkItemId` to link the Renovate PRs because of our branch policies as mentioned in the [docs](https://docs.renovatebot.com/configuration-options/#azureworkitemid). 

Renovate behavior is:
- Renovate does not update/change the linked work items for existing PRs

The ID is neither updated nor added to existing PRs.

I would rather expect this behavior (especially because PRs are updated automatically by renovate):
-Renovate updates the linked work item in PRs and removing others

Maybe also state it more out in the docs that it is only done on new PR creation, would help a lot.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Which of these use cases are you describing?
- PR was created without work item ID, config was updated later to add it, or
- work item ID changes in config while PR is open

**Reply by m-isenmann** [reply]
I was currently war running into it for the first use case "PR was created without work item ID, config was updated later to add it"
Still, I would also consider it for the 2nd use case. A open PR is updated by renovate and the work item link should also be updated in my opinion.

**Comment by rarkins** [comment]
Converted to issue: https://github.com/renovatebot/renovate/issues/32135
